pkg/operator/starter.go

```diff
@@ -28,133 +39,377 @@ const (
+				"rbac/prometheus_rolebinding.yaml",
+				"rbac/kube_rbac_proxy_role.yaml",
+				"rbac/kube_rbac_proxy_binding.yaml",
+			},
+			(&resourceapply.ClientHolder{}).WithKubernetes(controlPlaneKubeClient).WithDynamicClient(controlPlaneDynamicClient),
+			guestOperatorClient,
+			eventRecorder,
+		).AddKubeInformers(controlPlaneKubeInformersForNamespaces)
 
-	if err != nil {
-		return err
+		klog.Info("Starting static resources controller")
+		go staticResourcesController.Run(ctx, 1)
+
+		serviceMonitorController := staticresourcecontroller.NewStaticResourceController(
+			"PowerVSBlockCSIServiceMonitorController",
+			assets.ReadFile,
+			[]string{"servicemonitor.yaml"},
+			(&resourceapply.ClientHolder{}).WithDynamicClient(controlPlaneDynamicClient),
+			guestOperatorClient,
+			eventRecorder,
+		).WithIgnoreNotFoundOnCreate()
+
+		klog.Info("Starting ServiceMonitor controller")
+		go serviceMonitorController.Run(ctx, 1)
 	}
 
-	klog.Info("Starting the informers")
-	go kubeInformersForNamespaces.Start(ctx.Done())
-	go dynamicInformers.Start(ctx.Done())
-	go configInformers.Start(ctx.Done())
+	klog.Info("Starting control plane informers")
+	go controlPlaneKubeInformersForNamespaces.Start(ctx.Done())
+
+	klog.Info("Starting control plane controllerset")
+	go controlPlaneCSIControllerSet.Run(ctx, 1)
+
+	klog.Info("Starting guest cluster informers")
+	go guestKubeInformersForNamespaces.Start(ctx.Done())
+	go guestDynamicInformers.Start(ctx.Done())
+	go guestConfigInformers.Start(ctx.Done())
 
-	klog.Info("Starting controllerset")
-	go csiControllerSet.Run(ctx, 1)
+	klog.Info("Starting guest cluster controllerset")
+	go guestCSIControllerSet.Run(ctx, 1)
 
 	<-ctx.Done()
 
 	return fmt.Errorf("stopped")
 }
+
+func assetWithNamespaceFunc(namespace string) resourceapply.AssetFunc {
+	return func(name string) ([]byte, error) {
+		content, err := assets.ReadFile(name)
+		if err != nil {
+			panic(err)
+		}
+		return bytes.ReplaceAll(content, []byte("${NAMESPACE}"), []byte(namespace)), nil
+	}
+}
+
+func withNamespaceDeploymentHook(namespace string) dc.DeploymentHookFunc {
+	return func(_ *opv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		deployment.Namespace = namespace
+		return nil
+	}
+}
+
+func withHypershiftReplicasHook(isHypershift bool, guestNodeLister corev1listers.NodeLister) dc.DeploymentHookFunc {
+	if !isHypershift {
+		return csidrivercontrollerservicecontroller.WithReplicasHook(guestNodeLister)
+	}
+	return func(_ *opv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		// TODO: get this information from HostedControlPlane.Spec.AvailabilityPolicy
+		replicas := int32(1)
+		deployment.Spec.Replicas = &replicas
+		return nil
+	}
+}
+
+func withHypershiftDeploymentHook(isHypershift bool, hypershiftImage string) dc.DeploymentHookFunc {
+	return func(_ *opv1.OperatorSpec, deployment *appsv1.Deployment) error {
+		if !isHypershift {
+			return nil
+		}
+
+		deployment.Spec.Template.Spec.PriorityClassName = hypershiftPriorityClass
```

Comment:
Should we do same thing as https://github.com/openshift/aws-ebs-csi-driver-operator/pull/186 ?